qcodes/utils/installation_info.py

```diff
@@ -46,11 +47,40 @@ def is_qcodes_installed_editably() -> Optional[bool]:
     return answer
 
 
+def is_qcodes_installed_editable_fast() -> bool:
+    package_name = "qcodes"
+    # classical setuptools editable install
+    egg_link_found = any(
+        [
+            os.path.isfile(os.path.join(path, f"{package_name}.egg-link"))
+            for path in sys.path
+        ]
+    )
+    # pep 660 based install
+    dist_dirs = []
+    for path in sys.path:
+        dist_dirs.extend(glob.glob(f"{os.path.join(path, package_name)}*.dist-info"))
+    direct_urls = [
+        os.path.join(path, "direct_url.json")
+        for path in dist_dirs
+        if os.path.isfile(os.path.join(path, "direct_url.json"))
+    ]
+    # now check for dir_info value must be {'editable': true}.
+    return egg_link_found
+
+
 def get_qcodes_version() -> str:
     """
     Get the version of the currently installed QCoDeS
     """
-    return qcodes.__version__
+    if is_qcodes_installed_editable_fast():
+        import versioningit
+
+        pyprojectpath = Path(os.path.abspath(__file__)).parent.parent.parent
+        __version__ = versioningit.get_version(project_dir=pyprojectpath)
```

Comment:
should probably wrap this in a try except log in case the pyproject.toml file cannot be found for whatever reason